lib/HTML/FormBuilder/Validation.pm

```diff
@@ -668,6 +687,46 @@ validate form input and return true or false
 
 check the erorr is founded in the input element or not
 
+=head1 CROSS SITE REQUEST FORGERY PROTECTION
+
+for plain CGI or other framework, read Dancer example below.
+
+=head2 CSRF and Dancer
+
+=over 4
+
+=item * create form HTML and store csrftoken in session
+
+    my $form = HTML::FormBuilder::Validation->new(data => $form_attributes, csrftoken => 1);
+    ...
+    my $html = $form->build;
+
+    # save csrf token in session or cookie
+    session(__csrftoken => $form->csrftoken);
+
+=item * validate csrftoken on form submit
+
+    my $csrftoken = session('__csrftoken');
+    my $form = HTML::FormBuilder::Validation->new(data => $form_attributes, csrftoken => $csrftoken);
+    $form->validate_csrf() or die 'CSRF failed.';
+    # or call
+    if ( $form->validate() ) { # it calls validate_csrf inside
+        # Yap! it's ok
+    } else {
+        # NOTE we do not have error for csrf on form HTML build
+        # show form again with $form->build
+    }
+
+=back
+
+=head2 CSRF and Mojolicious
+
+if you're using L<Mojolicious> and have DefaultHelpers plugin enabled, it's simple to add csrftoken in Validation->new as below:
+
+    my $form = HTML::FormBuilder::Validation->new(data => $form_attributes, csrftoken => $c->csrf_token);
+
```

Comment:
Mojo saves to a signed cookie for session. so that's a bit limited but enough for now.